selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/stack/CreateWorkspaceFromJavaCentosStackTest.java

```diff
@@ -0,0 +1,140 @@
+import static org.eclipse.che.selenium.pageobject.dashboard.NewWorkspace.Stack.JAVA_CENTOS;
+import static org.openqa.selenium.Keys.ENTER;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import java.util.List;
+import org.eclipse.che.selenium.core.client.TestWorkspaceServiceClient;
+import org.eclipse.che.selenium.core.user.DefaultTestUser;
+import org.eclipse.che.selenium.pageobject.CheTerminal;
+import org.eclipse.che.selenium.pageobject.Consoles;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.eclipse.che.selenium.pageobject.dashboard.CreateWorkspaceHelper;
+import org.eclipse.che.selenium.pageobject.dashboard.Dashboard;
+import org.openqa.selenium.By;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/** @author Skoryk Serhii */
+public class CreateWorkspaceFromJavaCentosStackTest {
+
+  private static final String WORKSPACE_NAME = generate("workspace", 4);
+  private static final String CONSOLE_JAVA_SIMPLE = "console-java-simple";
+  private static final String WEB_JAVA_SPRING = "web-java-spring";
+
+  private List<String> projects = ImmutableList.of(CONSOLE_JAVA_SIMPLE, WEB_JAVA_SPRING);
+
+  @Inject private Ide ide;
+  @Inject private Consoles consoles;
+  @Inject private Dashboard dashboard;
+  @Inject private CheTerminal terminal;
+  @Inject private DefaultTestUser defaultTestUser;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private CreateWorkspaceHelper createWorkspaceHelper;
+  @Inject private TestWorkspaceServiceClient workspaceServiceClient;
+
+  @BeforeClass
+  public void setUp() {
+    dashboard.open();
+  }
+
+  @AfterClass
+  public void tearDown() throws Exception {
+    workspaceServiceClient.delete(WORKSPACE_NAME, defaultTestUser.getName());
+  }
+
+  @Test
+  public void checkWorkspaceCreationFromJavaCentosStack() {
+    createWorkspaceHelper.createWorkspaceFromStackWithProjects(
+        JAVA_CENTOS, WORKSPACE_NAME, projects);
+
+    ide.switchToIdeAndWaitWorkspaceIsReadyToUse();
+
+    projectExplorer.waitProjectInitialization(CONSOLE_JAVA_SIMPLE);
+    projectExplorer.waitProjectInitialization(WEB_JAVA_SPRING);
+  }
+
+  @Test(priority = 1)
+  public void checkConsoleJavaSimpleProjectCommands() {
+    consoles.executeCommandFromProjectExplorer(
+        CONSOLE_JAVA_SIMPLE, BUILD_GOAL, BUILD_COMMAND, BUILD_SUCCESS);
+
+    consoles.executeCommandFromProjectExplorer(
+        CONSOLE_JAVA_SIMPLE,
+        BUILD_GOAL,
+        BUILD_COMMAND_ITEM.getItem(CONSOLE_JAVA_SIMPLE),
+        BUILD_SUCCESS);
+
+    consoles.executeCommandFromProjectExplorer(
+        CONSOLE_JAVA_SIMPLE,
+        RUN_GOAL,
+        RUN_COMMAND_ITEM.getItem(CONSOLE_JAVA_SIMPLE),
+        "Hello World Che!");
+  }
+
+  @Test(priority = 1)
+  public void checkWebJavaSpringProjectCommands() {
+    consoles.executeCommandFromProjectExplorer(
```

Comment:
It's not quite clear what next sequences of command are doing exactly. Could you, please, comment it in the code?